cmd/zfs_object_agent/src/object_access.rs

```diff
@@ -79,6 +101,28 @@ where
         tokio::time::sleep(delay).await;
         delay = delay.mul_f64(thread_rng().gen_range(1.5..2.5));
     };
+    result
+}
+
+async fn retry<F, O, E>(msg: &str, d: Option<Duration>, f: impl Fn() -> F) -> Result<O, OAError<E>>
+where
+    E: core::fmt::Debug + core::fmt::Display + std::error::Error,
+    F: Future<Output = (bool, Result<O, E>)>,
+{
+    debug!("{}: begin", msg);
+    let begin = Instant::now();
+    let result = match d {
+        Some(timeout) => match future::timeout(timeout, retry_impl(msg, f)).await {
+            Err(e) => Err(OAError::TimeoutError { e: e }),
+            Ok(res2) => match res2 {
+                Ok(o) => Ok(o),
+                Err(e) => Err(OAError::RequestError { e: e }),
+            },
```

Comment:
I think you can use map_err() as you do below?
```suggestion
            Ok(res2) => res2.map_err(|e| OAError::RequestError { e }),
```

Reply:
For some reason I thought this didn't work, but it does.